cache/refs.go

```diff
@@ -1099,8 +1099,10 @@ func (sr *immutableRef) prepareRemoteSnapshotsStargzMode(ctx context.Context, s
 					if err == nil { // usable as remote snapshot without unlazying.
 						defer func() {
 							// Remove tmp labels appended in this func
-							for k := range tmpLabels {
-								info.Labels[k] = ""
+							if info.Labels != nil {
```

Comment:
Could you add a debug log saying snapshot already exists but tmp remote labels are nil?

Reply:
Thanks @ktock!

Added in [395e72c](https://github.com/moby/buildkit/pull/4187/commits/395e72c52b12f5e87ba4149f3ec559a02600de20)